src/backend/aspen/api/views/phylo_runs.py

```diff
@@ -269,3 +271,28 @@ async def delete_run(
     await db.delete(item)
     await db.commit()
     return PhyloRunDeleteResponse(id=item_db_id)
+
+
+@router.put("/{item_id}")
+async def update_phylo_tree_and_run(
+    item_id: int,
+    phylo_run_update_request: PhyloRunUpdateRequest,
+    db: AsyncSession = Depends(get_db),
+    user: User = Depends(get_auth_user),
+):
+    # get phylo_run and check that user has permission to update PhyloRun/PhyloTree
+    phylo_run = await get_editable_phylo_run_by_id(db, item_id, user)
+
+    async with db as session:
```

Comment:
`db` is already an `AsyncSession` -- do we need this `with` ?

Reply:
yes because otherwise we get an io error when updating these objects.

Reply:
I don't think we do ... this should work without the `with` and just running `await db.commit()` below.

Reply:
removed